packages/contracts/src.ts/index.ts

```diff
@@ -1236,10 +1224,10 @@ export class ContractFactory {
         params.push(overrides);
 
         // Get the deployment transaction (with optional overrides)
-        const unsignedTx = this.getDeployTransaction(...params);
+        const unsignedTx = this.getDeployTransactions();
```

Comment:
We should pass the `...params` otherwise we won't be able to deploy a contract with constructor arguments or pass the `gasLimit`